chainer/_link.py

```diff
@@ -0,0 +1,271 @@
+import typing as tp  # NOQA
+
+import chainer
+from chainer import backend
+from chainer.backends import cuda
+from chainer.backends import intel64
+from chainer import types  # NOQA
+import chainerx
+
+
+class DeviceObjectsVisitableMixin(object):
```

Comment:
I want some new concept and a name for it that indicates "a class/object that supports the canonical device identification/transfer APIs" if you expect that this feature will be public and used by non-link classes.

Reply:
Yes, I want to make it more easily understood concept than "device objects visitable", but I don't come up with a better name/generalization.

Reply:
Some random ideas
* `DeviceObject`
* `DeviceResident`
* `DeviceOccupant`